tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py

```diff
@@ -1254,101 +1273,259 @@ def test_seed_model_raises(self):
-            ('Estimator', sklearn.naive_bayes.GaussianNB())]
-        )
-        task = openml.tasks.get_task(20)
-        train, test = task.get_train_test_split_indices(repeat_nr, fold_nr)
-        X, y = task.get_X_and_y()
-        clf.fit(X[train], y[train])
-
-        test_X = X[test]
-        test_y = y[test]
-
-        probaY = clf.predict_proba(test_X)
-        predY = clf.predict(test_X)
-        sample_nr = 0  # default for this task
-        for idx in range(0, len(test_X)):
-            arff_line = self.extension._prediction_to_row(
-                rep_no=repeat_nr,
-                fold_no=fold_nr,
-                sample_no=sample_nr,
-                row_id=idx,
-                correct_label=task.class_labels[test_y[idx]],
-                predicted_label=predY[idx],
-                predicted_probabilities=probaY[idx],
-                class_labels=task.class_labels,
-                model_classes_mapping=clf.classes_,
+                                            task_type=task.task_type_id, check_scores=False)
+
+    def test_run_model_on_fold_classification_3(self):
+
+        class HardNaiveBayes(sklearn.naive_bayes.GaussianNB):
+            # class for testing a naive bayes classifier that does not allow soft
+            # predictions
+            def __init__(self, priors=None):
+                super(HardNaiveBayes, self).__init__(priors)
+
+            def predict_proba(*args, **kwargs):
+                raise AttributeError('predict_proba is not available when '
+                                     'probability=False')
+
+        # task 1 (test server) is important: it is a task with an unused class
+        tasks = [1, 3, 115]
+        flow = unittest.mock.Mock()
+        flow.name = 'dummy'
+
+        for task_id in tasks:
+            task = openml.tasks.get_task(task_id)
+            X, y = task.get_X_and_y()
+            train_indices, test_indices = task.get_train_test_split_indices(
+                repeat=0, fold=0, sample=0)
+            X_train = X[train_indices]
+            y_train = y[train_indices]
+            X_test = X[test_indices]
+            clf1 = sklearn.pipeline.Pipeline(steps=[
+                ('imputer', sklearn.preprocessing.Imputer()),
+                ('estimator', sklearn.naive_bayes.GaussianNB())
+            ])
+            clf2 = sklearn.pipeline.Pipeline(steps=[
+                ('imputer', sklearn.preprocessing.Imputer()),
+                ('estimator', HardNaiveBayes())
+            ])
+
+            pred_1, proba_1, _, _ = self.extension._run_model_on_fold(
+                model=clf1,
+                task=task,
+                X_train=X_train,
+                y_train=y_train,
+                X_test=X_test,
+                fold_no=0,
+                rep_no=0,
+                classes=task.class_labels,
+            )
+            pred_2, proba_2, _, _ = self.extension._run_model_on_fold(
```

Comment:
`proba_1` and `proba_2` are unused? so no need to name them either.
You can ignore any amount of return arguments with `pred_1, *_`.

Reply:
I added some additional asserts to actually use them.